starknet_py/utils/typed_data.py

```diff
@@ -130,26 +141,34 @@ def _encode_value(
         if is_pointer(type_name) and isinstance(value, list):
             type_name = strip_pointer(type_name)
             hashes = [self._encode_value(type_name, val) for val in value]
-            return compute_hash_on_elements(hashes)
+            return self._hash_method.hash_many(hashes)
```

Comment:
This is a fix - we should hash using `hash_method.hash_many()` (either Pedersen or Poseidon) , not directly call `compute_hash_on_elements` which uses Pedersen hash.